generators/rest-adapter-core/src/test/java/uk/gov/justice/services/adapter/rest/filter/JsonValidatorRequestFilterTest.java

```diff
@@ -0,0 +1,133 @@
+package uk.gov.justice.services.adapter.rest.filter;
+
+import static java.util.UUID.randomUUID;
+import static javax.ws.rs.core.MediaType.valueOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.justice.services.common.http.HeaderConstants.ID;
+import static uk.gov.justice.services.common.http.HeaderConstants.USER_ID;
+
+import uk.gov.justice.services.adapter.rest.exception.BadRequestException;
+import uk.gov.justice.services.messaging.logging.HttpTraceLoggerHelper;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.slf4j.Logger;
+
+@RunWith(MockitoJUnitRunner.class)
+public class JsonValidatorRequestFilterTest {
+
+    private static final MediaType TEST_JSON_MEDIA_TYPE = valueOf("application/vnd.test.action.name+json");
+
+    @Mock
+    private Logger logger;
+
+    @Mock
+    private HttpTraceLoggerHelper httpTraceLoggerHelper;
+
+    @InjectMocks
+    private JsonValidatorRequestFilter filterUnderTest;
+
+    @Captor
+    private ArgumentCaptor<InputStream> captor = ArgumentCaptor.forClass(InputStream.class);
```

Comment:
You  don't need ArgumentCaptor.forClass(InputStream.class) if you use the @Captor annotation

Reply:
Done